game/tests/test_level_editor.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Code for Life
+#
+# Copyright (C) 2016, Ocado Innovation Limited
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# ADDITIONAL TERMS – Section 7 GNU General Public Licence
+#
+# This licence does not grant any right, title or interest in any “Ocado” logos,
+# trade names or the trademark “Ocado” or any other trademarks or domain names
+# owned by Ocado Innovation Limited or the Ocado group of companies or any other
+# distinctive brand features of “Ocado” as may be secured from time to time. You
+# must not distribute any modification of this program using the trademark
+# “Ocado” or claim any affiliation or association with Ocado or its employees.
+#
+# You are not authorised to use the name Ocado (or any of its trade names) or
+# the names of any author or contributor in advertising or for publicity purposes
+# pertaining to the distribution of this program, without the prior written
+# authorisation of Ocado.
+#
+# Any propagation, distribution or conveyance of this program must include this
+# copyright notice and these terms. You must not misrepresent the origins of this
+# program; modified versions of the program must be marked as such and not
+# identified as the original program.
+import json
+
+from django.core.urlresolvers import reverse
+from django.test.testcases import TestCase
+from django.test.client import Client
+
+from hamcrest import *
+
+from game.tests.utils.teacher import signup_teacher_directly, create_school, add_teacher_to_school
+from game.tests.utils.level import create_save_level
+
+
+class LevelEditorTestCase(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+
+    def login(self, email, password):
+        self.client.post(reverse('teach'), {'login-email': email, 'login-password': password, 'login': ''}, follow=True)
+
+    def get_sharing_information(self, level_id):
+        url = reverse('get_sharing_information_for_editor', args=[level_id])
+        response = self.client.get(url)
+        return response
+
+    def test_level_sharing(self):
```

Comment:
I think this would be better off being split into smaller, more granular individual test methods e.g.: `test_level_sharing_with_no_school`, `test_level_sharing_with_school` (which would be lines 73-78), `test_level_sharing_with_empty_school`.